readthedocs/templates/404.html

```diff
@@ -16,6 +16,46 @@
 {% block language-select-form %}{% endblock %}
 
 {% block content %}
+
+<h1>{% trans "404 - page not found" %}</h1>
+
+{% if project %}
```

Comment:
Since we are raising custom exceptions now, I'd suggest to raise a _specific exception for each particular case_ (e.g. `ProxitoHttp404ProjectNotFound`, `ProxitoHttp404VersionNotFound`, etc) instead one generic `ProxitoHttp404` and remove all this logic from the template.

That way, we have all the logic in Python and we can render different 404 templates with only the content they need. It also simplifies the reading of this code, since we don't have to follow Python and Django template logic to understand what's the final response.

Reply:
I agree - that will mean the contextualization can also scale quite a lot if needed... in terms of both user-facing messages and the view logic such as notifications, automated redirects, error logging etc.

I'll have to _not_ respect the `template_name` keyword argument for the 404 view. But I think that's fine, we can still use template inheritance from one main `404.html` base template.

Reply:
Very cool suggestion, I've added a note to also handle "version not found". The templates are now being split up and new exception classes are being added.